RecoEgamma/EgammaHLTProducers/plugins/EgammaHLTHcalVarProducerFromRecHit.cc

```diff
@@ -32,12 +32,16 @@ A rho correction can be applied
 #include "RecoLocalCalo/HcalRecAlgos/interface/HcalSeverityLevelComputerRcd.h"
 #include "CondFormats/HcalObjects/interface/HcalChannelQuality.h"
 #include "CondFormats/DataRecord/interface/HcalChannelQualityRcd.h"
+#include "CondFormats/DataRecord/interface/HcalPFCutsRcd.h"
+#include "CondTools/Hcal/interface/HcalPFCutsHandler.h"
+#include "Geometry/CaloTopology/interface/HcalTopology.h"
 
 class EgammaHLTHcalVarProducerFromRecHit : public edm::global::EDProducer<> {
 public:
   explicit EgammaHLTHcalVarProducerFromRecHit(const edm::ParameterSet &);
 
 public:
+  void beginRun(edm::Run const &, edm::EventSetup const &);
```

Comment:
we overlooked that this module is a `edm::global::EDProducer` hence `beginRun` cannot be `override`-n (it's not in the base class). Hence this is never called.

Reply:
The best fix would be to move the code in `beginRun()` to `produce()` and remove the `hcalCuts` member variable.

Reply:
https://github.com/cms-sw/cmssw/pull/43819 should fix.